indico/modules/events/registration/util.py

```diff
@@ -185,3 +185,17 @@ def generate_csv_from_registrations(registrations, regform_items, special_items)
         writer.writerow(registration_dict)
     buf.seek(0)
     return buf
+
+
+def get_user_complete_registrations(user, event):
+    return Registration.find(Registration.user == user,
+                             RegistrationForm.event_id == event.id,
+                             ~Registration.is_deleted,
+                             ~RegistrationForm.is_deleted,
+                             Registration.state == RegistrationState.complete,
+                             _join=Registration.registration_form)
+
+
+def get_registrations_with_tickets(user, event):
+    return get_user_complete_registrations(user, event).filter(RegistrationForm.tickets_enabled,
```

Comment:
Wrong logic. multiple filter args are combined with AND`. you want `RegistrationForm.tickets_enabled | RegistrationForm.ticket_on_event_page`